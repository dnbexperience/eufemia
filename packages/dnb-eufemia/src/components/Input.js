/**
 * ATTENTION: This file is auto generated by using "prepareTemplates".
 * Do not change the content!
 *
 */

/**
 * Library Index input to autogenerate all the components and patterns
 * Used by "prepareInputs"
 */

import Input from './input/Input'
export * from './input/Input'
export default Input
