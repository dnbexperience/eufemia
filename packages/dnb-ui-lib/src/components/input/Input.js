/**
 * Web Input Component
 *
 */

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { DefaultIconSize } from '../icon'
import Button from '../button/Button'
import FormLabel from '../form-label/FormLabel'
import FormStatus from '../form-status/FormStatus'
import {
  registerElement,
  validateDOMAttributes,
  processChildren,
  pickRenderProps,
  dispatchCustomElementEvent
} from '../../shared/component-helper'

const renderProps = {
  on_change: null,
  on_submit: null,
  on_focus: null,
  on_blur: null,
  on_submit_focus: null,
  on_submit_blur: null
}

export const propTypes = {
  type: PropTypes.string,
  size: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  id: PropTypes.string,
  label: PropTypes.string,
  status: PropTypes.string,
  status_state: PropTypes.string,
  autocomplete: PropTypes.oneOf(['on', 'off']),
  search_button_title: PropTypes.string,
  placeholder: PropTypes.string,
  description: PropTypes.string,
  align: PropTypes.string,
  extra_information: PropTypes.string,
  disabled: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  class: PropTypes.string,
  input_class: PropTypes.string,
  input_typo: PropTypes.string,
  attributes: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),

  // React props
  className: PropTypes.string,
  inputElement: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.func
  ]),

  // Web Component props
  custom_element: PropTypes.object,
  custom_method: PropTypes.func,
  on_change: PropTypes.func,
  on_submit: PropTypes.func,
  on_focus: PropTypes.func,
  on_blur: PropTypes.func,
  on_submit_focus: PropTypes.func,
  on_submit_blur: PropTypes.func
}

export const defaultProps = {
  type: 'text',
  size: null,
  value: null,
  id: null,
  label: null,
  status: null,
  status_state: 'error',
  autocomplete: 'off',
  search_button_title: '',
  placeholder: null,
  description: null,
  align: null,
  extra_information: null,
  disabled: false,
  input_class: null,
  input_typo: 'dnb-typo-number--lining',
  class: null,
  attributes: null,

  // React props
  className: null,
  inputElement: null,
  children: null,

  // Web Component props
  custom_element: null,
  custom_method: null,
  ...renderProps
}

/**
 * The input component is an umbrella component for all inputs which share the same style as the classic `text` input field. Radio buttons and other form elements are not included here.
 */
export default class Input extends PureComponent {
  static tagName = 'dnb-input'
  static propTypes = propTypes
  static defaultProps = defaultProps
  static renderProps = renderProps

  static enableWebComponent() {
    registerElement(Input.tagName, Input, defaultProps)
  }

  static getDerivedStateFromProps(props, state) {
    const value = Input.getValue(props)
    if (
      state._listenForPropChanges &&
      value !== null &&
      value !== state.value
    ) {
      state.value = value
    }
    if (props.disabled) {
      state.inputState = 'disabled'
    }
    state._listenForPropChanges = true
    return state
  }

  static getValue(props) {
    if (props.value) return props.value
    return processChildren(props)
  }

  state = { inputState: 'virgin', value: null }

  constructor(props) {
    super(props)

    this._ref = React.createRef()
    this._id = props.id || `dnb-input-${Math.round(Math.random() * 999)}` // cause we need an id anyway

    // pass along all props we wish to have as params
    this.renderProps = pickRenderProps(props, renderProps)

    // make sure we dont trigger getDerivedStateFromProps on startup
    this.state._listenForPropChanges = true
    this.state.value = Input.getValue(props)

    // This is only to show, that we also can add a custom method to the custom element, and call it like so:
    // this._ref.current.focus() // current would be the Custom Element - we may delay it for 1 tick, to make sure we call if after it is set
    // We also have to ahve this method in here
    // setCustomElementMethod(this, 'focus', () => {
    //   setTimeout(() => {
    //     try {
    //       this._ref.current.focus()
    //     } catch (e) {
    //       //
    //     }
    //   }, 100)
    // })
  }
  onFocusHandler = event => {
    const value = event.target.value
    this.setState({
      value,
      _listenForPropChanges: false,
      inputState: 'focus'
    })
    dispatchCustomElementEvent(this, 'on_focus', { value, event })
  }
  onBlurHandler = event => {
    const value = event.target.value
    this.setState({
      value,
      _listenForPropChanges: false,
      inputState: 'dirty'
    })
    dispatchCustomElementEvent(this, 'on_blur', { value, event })
  }
  onChangeHandler = event => {
    const value = event.target.value
    this.setState({ value, _listenForPropChanges: false })
    event.preventDefault()
    dispatchCustomElementEvent(this, 'on_change', { value, event })
  }
  onKeyDownHandler = event => {
    if (event.key === 'Enter') {
      const value = event.target.value
      event.preventDefault()
      dispatchCustomElementEvent(this, 'on_submit', { value, event })
    }
  }
  render() {
    const {
      type,
      size,
      label,
      status,
      status_state,
      disabled,
      placeholder,
      description,
      align,
      input_class,
      input_typo,
      extra_information
    } = this.props

    let { autocomplete } = this.props
    if (type === 'search') {
      autocomplete = null
    }
    const id = this._id

    const classes = classnames(
      'dnb-input',
      `dnb-input--${type}`, //type_modifier
      extra_information ? 'dnb-input--has-extra-information' : null,
      size ? 'dnb-input--' + size : '',
      type === 'search' ? 'dnb-input__input--search' : null,
      align ? `dnb-input__align--${align}` : null,
      status ? `dnb-input__status--${status_state}` : null,
      this.props.class,
      this.props.className
    )

    const { inputElement: Elem, ...renderProps } = this.renderProps

    const inputParams = {
      ...renderProps,
      className: classnames('dnb-input__input', input_typo, input_class),
      autoComplete: autocomplete,
      value: this.state.value || '',
      type,
      id,
      // align,
      disabled,
      name: id,
      onChange: this.onChangeHandler,
      onKeyDown: this.onKeyDownHandler,
      onFocus: this.onFocusHandler,
      onBlur: this.onBlurHandler
    }

    // also used for code markup simulation
    validateDOMAttributes(this.props, inputParams)

    if (description) {
      inputParams['aria-describedby'] = id + '-description'
    }

    const shellParams = {
      'data-input-state': this.state.inputState,
      'data-has-content':
        String(this.state.value || '').length > 0 ? 'true' : 'false'
    }

    return (
      <span className={classes}>
        {label && (
          <FormLabel for_id={id} text={label} disabled={disabled} />
        )}

        <span className="dnb-input__shell" {...shellParams}>
          {(type === 'text' || type === 'number' || type === 'search') &&
            ((typeof Elem === 'function' ? (
              <Elem innerRef={this._ref} {...inputParams} />
            ) : null) || <input ref={this._ref} {...inputParams} />)}

          {type === 'search' && (
            <Submit
              {...this.props}
              value={inputParams.value}
              title={this.props.search_button_title}
            />
          )}

          {placeholder && (
            <span
              className={classnames(
                'dnb-input__placeholder',
                align ? `dnb-input__align--${align}` : null
              )}
            >
              {placeholder}
            </span>
          )}
        </span>

        {this.props.description && (
          <span
            className="dnb-input__description"
            id={id + '-description'}
          >
            {this.props.description}
          </span>
        )}

        {extra_information && (
          <span className="dnb-input__extra-information">
            {extra_information}
          </span>
        )}

        {status && status !== 'true' && (
          <FormStatus for_id={id} text={status} status={status_state} />
        )}
      </span>
    )
  }
}

class Submit extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    icon_size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

    // React props
    onSubmit: PropTypes.func,
    onSubmitFocus: PropTypes.func,
    onSubmitBlur: PropTypes.func,

    // Web Component props
    on_submit: PropTypes.func,
    on_submit_focus: PropTypes.func,
    on_submit_blur: PropTypes.func
  }

  static defaultProps = {
    title: null,
    icon_size: DefaultIconSize,

    // React props
    onSubmit: null,
    onSubmitFocus: null,
    onSubmitBlur: null,

    // Web Component props
    on_submit: null,
    on_submit_focus: null,
    on_submit_blur: null
  }

  state = { focusState: 'virgin' }

  onFocusHandler = event => {
    const value = this.props.value
    this.setState({
      focusState: 'focus'
    })
    dispatchCustomElementEvent(this, 'on_submit_focus', { value, event })
  }
  onBlurHandler = event => {
    const value = this.props.value
    this.setState({
      focusState: 'dirty'
    })
    dispatchCustomElementEvent(this, 'on_submit_blur', { value, event })
  }
  onSubmitHandler = event => {
    const value = this.props.value
    dispatchCustomElementEvent(this, 'on_submit', { value, event })
  }
  render() {
    const { title } = this.props
    const params = {
      type: 'submit',
      title
    }

    // also used for code markup simulation
    validateDOMAttributes(this.props, params)

    return (
      <span
        className="dnb-input__search-submit"
        data-input-state={this.state.focusState}
      >
        <Button
          className="dnb-input__search-submit__button"
          variant="secondary"
          onClick={this.onSubmitHandler}
          onFocus={this.onFocusHandler}
          onBlur={this.onBlurHandler}
          icon="search"
          {...params}
        />
      </span>
    )
  }
}
