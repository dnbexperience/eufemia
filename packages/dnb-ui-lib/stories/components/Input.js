/**
 * dnb-ui-lib Component Story
 *
 */

import React, { useState, useEffect } from 'react'
import { Wrapper, Box } from '../helpers'
import styled from '@emotion/styled'

import {
  Input,
  InputMasked,
  Button,
  FormSet,
  FormRow,
  FormLabel
} from '../../src/components'

const CustomStyle = styled.div`
  p {
    background-color: rgba(213, 30, 149, 0.25);
  }
`

export default [
  'Inputs',
  () => (
    <CustomStyle>
      <Wrapper>
        <Box>
          <Input
            label="Choose file:"
            type="file"
            accept="image/png, image/jpeg"
          />
        </Box>
        <Box>
          <CustomInput />
        </Box>
        <Box>
          <FormSet>
            <FormRow
              indent
              indent_offset="x-large"
              wrap
              label="Long label labwl Adipiscing mauris dis proin nec Condimentum egestas class blandit netus non a suscipit id urna:"
            >
              <Input label="Input A:" top="small" right="small" />
              <Input label="Input B:" top="small" right="small" />
              <Input label="Input C:" top="small" right="small" />
            </FormRow>
          </FormSet>
        </Box>
        <Box>
          <FormRow
            label="Vertical label:"
            label_direction="vertical"
            // vertical
          >
            <Input label="Input label A:" right="small" />
            <Input label="Input label B:" />
          </FormRow>
        </Box>
        <Box>
          <FormSet direction="vertical">
            <FormRow label="Legend:">
              <Input label="Vertical 1:" />
              <Input label="Vertical 2:" stretch top="small" />
            </FormRow>
          </FormSet>
        </Box>
        <Box>
          <FormSet vertical>
            <FormRow label="Legend:">
              <Input label="Vertical 1:" />
              <Input label="Vertical 2:" stretch top="small" />
            </FormRow>
          </FormSet>
        </Box>
        <Box>
          <Input
            label="Vertical label:"
            value="Stretch me ..."
            stretch
            label_direction="vertical"
          />
        </Box>
        <Box>
          Text
          <Input
            selectall
            label="Label:"
            on_change={event => {
              console.log('on_change', event)
            }}
          >
            Input ...
          </Input>
          Text
        </Box>
        <Box>
          Text
          <FormLabel>FormLabel:</FormLabel>
          <Input>Input ...</Input>
          Text
        </Box>
        <Box>
          <p className="dnb-p">
            <Input
              label="ReadOnly:"
              placeholder="Placeholder ..."
              readOnly
            />
          </p>
        </Box>
        <Box>
          <Input
            label="Search:"
            type="search"
            align="right"
            stretch
            submit_button_title="Search"
            placeholder="Search text placeholder"
          />
          <Input
            label="Search:"
            size="medium"
            type="search"
            align="right"
            stretch
            placeholder="Medium input with right aligned text"
          />
          <Input
            label="Search:"
            size="large"
            type="search"
            align="right"
            stretch
            placeholder="Large input with right aligned text"
          />
        </Box>
        <Box>
          <Input
            disabled
            status="Error"
            label="Disabled search:"
            type="search"
            submit_button_title="Search"
            placeholder="Search text placeholder"
          />
        </Box>
        <Box>
          <Input
            label="Input with status:"
            status="Message to the user"
            value="Input value with status"
          />
        </Box>
        <Box>
          <Input
            label="Input with description:"
            description="Description to the user"
            value="Input value with status"
          />
        </Box>
        <Box>
          <InputMasked
            label="Masked:"
            autocomplete="off"
            // value="1000000"
            mask={[
              '+',
              /[4]/, // have to start with 4
              /[5-7]/, // can be 5,6 or 7
              ' ',
              '/',
              ' ',
              /[49]/, // have to start with 4 or 9
              /\d/,
              /\d/,
              ' ',
              /\d/,
              /\d/,
              ' ',
              /\d/,
              /\d/,
              /\d/
            ]}
            show_mask="true"
          />
        </Box>
        <Box>
          <form
            onSubmit={event => {
              // console.log('onSubmit', event)
              event.preventDefault()
              // event.persist()
            }}
          >
            <InputUpdate />
            <Button
              text="Submit"
              type="submit"
              on_click={event => {
                console.log('on_click', event)
              }}
              onClick={event => {
                console.log('onClick', event)
              }}
            />
          </form>
        </Box>
      </Wrapper>
    </CustomStyle>
  )
]

const InputUpdate = () => {
  const [initValue, setNewValue] = useState('Input ...')

  useEffect(() => {
    const timeoutId = setTimeout(() => {
      setNewValue('')
    }, 1e3)
    return () => clearTimeout(timeoutId)
  })

  return (
    <Input
      label="Label:"
      on_change={({ value }) => {
        console.log('on_change', value)
      }}
      onChange={({ value }) => {
        console.log('onChange', value)
      }}
      on_submit={({ value }) => {
        console.log('on_submit', value)
      }}
      onSubmit={({ value }) => {
        console.log('onSubmit', value)
      }}
      value={initValue}
    />
  )
}

const CustomInput = () => {
  const [value, setValue] = useState('2019-02-15')
  return (
    <>
      <Input
        value={value}
        on_change={({ value }) => {
          console.log('on_change', value)
          setValue(value)
        }}
        on_state_update={({ value }) => {
          console.warn('on_state_update', value)
          setValue(value)
        }}
        right
      />
      <Button
        text="Reset"
        on_click={() => {
          setValue('123')
        }}
      />
    </>
  )
}
