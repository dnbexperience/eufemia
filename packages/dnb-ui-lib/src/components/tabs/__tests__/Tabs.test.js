/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../Tabs'
import Input from '../../Input'

const props = fakeProps(require.resolve('../Tabs'), {
  all: true,
  optional: true
})
delete props.render
props.id = 'id'

const startup_selected_key = 'second'
const tablistData = [
  { title: 'First', key: 'first' },
  { title: 'Second', key: 'second' },
  { title: 'Third', key: 'third' }
]
const tablistDataWithContent = [
  { title: 'First', key: 1, content: <h2>First</h2> }, // without function
  { title: 'Second', key: 2, content: () => <h2>Second</h2> }, // with function
  { title: 'Third', key: 3, content: () => <h2>Third</h2> } // with function
]
const contentWrapperData = {
  first: <h2>First</h2>, // without function
  second: () => <h2>Second</h2>, // with function
  third: <h2>Third</h2> // without function
}

describe('Tabs component', () => {
  const Comp = mount(
    <Component
      {...props}
      data={tablistData}
      selected_key={startup_selected_key}
    >
      {contentWrapperData}
    </Component>
  )

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('have a "selected_key" state have to be same as prop from startup', () => {
    expect(Comp.state().selected_key).toBe(startup_selected_key)
  })

  it('should validate with ARIA rules as a tabs', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('TabList component', () => {
  const Comp = mount(
    <Component
      {...props}
      data={tablistData}
      selected_key={startup_selected_key}
    >
      {contentWrapperData}
    </Component>
  )

  it('has to have the right amount of renderet components', () => {
    expect(Comp.find('button').length).toBe(tablistData.length)
    expect(Comp.find('div[role="tabpanel"]').length).toBe(1)
  })

  it('has to have the right content on a "click event"', () => {
    Comp.find('button[data-tab-key="third"]').simulate('click')
    expect(Comp.state().selected_key).toBe(tablistData[2].key) // get the third key
    expect(Comp.find('div[role="tabpanel"]').children().html()).toBe(
      mount(contentWrapperData.third).html()
    )
  })
})

describe('A single Tab component', () => {
  const Comp = mount(
    <Component
      {...props}
      data={tablistData}
      selected_key={startup_selected_key}
    >
      {contentWrapperData}
    </Component>
  )

  it('has to have a role="tab" attribute and a selcted class', () => {
    expect(
      Comp.find('button[data-tab-key="second"]')
        .instance()
        .getAttribute('role')
    ).toBe('tab')
    expect(
      Comp.find('button[data-tab-key="second"]').hasClass('selected')
    ).toBe(true)
  })

  it('has to have the right content on a keydown "ArrowRight"', () => {
    // reset the state
    Comp.find('button[data-tab-key="second"]').simulate('click')
    Comp.find('div[role="tablist"]').simulate('keyDown', {
      key: 'ArrowRight',
      keyCode: 39
    })
    expect(Comp.find('div[role="tabpanel"]').children().html()).toBe(
      mount(contentWrapperData.third).html()
    )
  })

  it('has to work with "data only" property containing a "content"', () => {
    const Comp = mount(<Component data={tablistDataWithContent} />)
    expect(Comp.find('button').first().hasClass('selected')).toBe(true)
    expect(Comp.find('div.dnb-tabs__content').text()).toBe('First')

    // then click on tab two
    // also test the ability of having a integer only as the key
    Comp.find('button[data-tab-key=2]').simulate('click')
    expect(Comp.find('div.dnb-tabs__content').text()).toBe('Second')
  })

  it('has to run "prevent_rerender" as supposed', () => {
    const Comp = mount(
      <Component
        prevent_rerender
        data={[
          {
            title: 'One',
            key: 'one',
            content: () => (
              <Input label="Content one" placeholder="Edit me" />
            )
          },
          { title: 'Two', key: 'two', content: 'Content two' }
        ]}
      />
    )

    expect(Comp.find('div.dnb-tabs__cached').exists()).toBe(true)

    // also check a real live rerender scenario
    const value = 'value'
    Comp.find('.dnb-input__input').simulate('change', {
      target: { value }
    })

    // then click on tab two
    Comp.find('button[data-tab-key="two"]').simulate('click')

    // the first cache should now be hidden
    expect(
      Comp.find('div.dnb-tabs__cached')
        .at(0)
        .instance()
        .getAttribute('aria-hidden')
    ).toBe('true')

    // and on tab one again
    Comp.find('button[data-tab-key="one"]').simulate('click')

    // the entered value should still be the same
    expect(Comp.find('.dnb-input__input').instance().value).toBe(value)

    expect(
      Comp.find('div.dnb-tabs__cached')
        .at(0)
        .instance()
        .getAttribute('aria-hidden')
    ).not.toBe('true')
    expect(
      Comp.find('div.dnb-tabs__cached')
        .at(1)
        .instance()
        .getAttribute('aria-hidden')
    ).toBe('true')
  })

  it('has to run "prerender" as supposed', () => {
    const Comp = mount(
      <Component
        prerender
        data={[
          {
            title: 'One',
            key: 1,
            content: 'Content one'
          },
          { title: 'Two', key: 2, content: 'Content two' }
        ]}
      />
    )

    expect(Comp.find('div.dnb-tabs__cached').exists()).toBe(true)

    expect(
      Comp.find('div.dnb-tabs__cached')
        .at(0)
        .instance()
        .hasAttribute('aria-hidden')
    ).toBe(false)
    expect(
      Comp.find('div.dnb-tabs__cached')
        .at(1)
        .instance()
        .getAttribute('aria-hidden')
    ).toBe('true')

    expect(Comp.find('div.dnb-tabs__cached').at(0).text()).toBe(
      'Content one'
    )
    expect(Comp.find('div.dnb-tabs__cached').at(1).text()).toBe(
      'Content two'
    )
  })

  it('has to work with "Tabs.Content" as children Components', () => {
    const Comp = mount(
      <Component data={tablistData}>
        <Component.Content title="first title">first</Component.Content>
        <Component.Content title="second title" selected>
          second
        </Component.Content>
      </Component>
    )
    expect(
      Comp.find('button.selected').instance().getAttribute('data-tab-key')
    ).toBe('second-title')
    expect(
      Comp.find('button').at(1).instance().getAttribute('data-tab-key')
    ).toBe('second-title')
    expect(Comp.find('div.dnb-tabs__content').text()).toBe('second')
    expect(
      Comp.find(
        'button[aria-selected=true] span.dnb-tabs__button__title'
      ).text()
    ).toBe('second title')
  })
})

describe('Tabs scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-tabs.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-tabs-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
