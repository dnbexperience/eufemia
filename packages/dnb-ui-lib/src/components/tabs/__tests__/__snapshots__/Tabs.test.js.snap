// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component have to match snapshot 1`] = `
<Tabs
  0={
    Object {
      "displayName": "Tabs",
      "props": Object {
        "align": "'left'",
        "children": Object {},
        "class": "class",
        "className": "className",
        "data": "data",
        "label": "label",
        "render": [Function],
        "selected_key": "selected_key",
        "use_hash": "use_hash",
      },
    }
  }
  1={
    Object {
      "displayName": "TabContent",
      "props": Object {
        "children": "children",
        "id": "id",
        "selection_key": "selection_key",
      },
    }
  }
  align="left"
  class={null}
  className={null}
  data={
    Array [
      Object {
        "key": "first",
        "title": "First",
      },
      Object {
        "key": "second",
        "title": "Second",
      },
      Object {
        "key": "third",
        "title": "Third",
      },
    ]
  }
  id="id"
  label={null}
  render={null}
  selected_key="second"
  use_hash={false}
>
  <Wrapper>
    <div
      className="dnb-tabs"
    >
      <TabsList>
        <div
          className="dnb-tabs__tabs dnb-tabs__tabs--left"
        >
          <Tabs>
            <div
              className="dnb-tabs__tabs__tablist dnb-tab-focus"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              role="tablist"
              tabIndex="0"
            >
              <button
                aria-selected={false}
                className="dnb-tablink dnb-no-mouse-focus tab--first "
                disabled={false}
                id="id-tab-first"
                onClick={[Function]}
                role="tab"
                tabIndex="-1"
              >
                <span
                  className="dnb-tablink-title"
                >
                  First
                </span>
                <span
                  aria-hidden={true}
                  hidden={true}
                >
                  First
                </span>
              </button>
              <button
                aria-controls="id-content-second"
                aria-selected={true}
                className="dnb-tablink dnb-no-mouse-focus tab--second selected"
                disabled={false}
                id="id-tab-second"
                onClick={[Function]}
                role="tab"
                tabIndex="-1"
              >
                <span
                  className="dnb-tablink-title"
                >
                  Second
                </span>
                <span
                  aria-hidden={true}
                  hidden={true}
                >
                  Second
                </span>
              </button>
              <button
                aria-selected={false}
                className="dnb-tablink dnb-no-mouse-focus tab--third "
                disabled={false}
                id="id-tab-third"
                onClick={[Function]}
                role="tab"
                tabIndex="-1"
              >
                <span
                  className="dnb-tablink-title"
                >
                  Third
                </span>
                <span
                  aria-hidden={true}
                  hidden={true}
                >
                  Third
                </span>
              </button>
            </div>
          </Tabs>
        </div>
      </TabsList>
      <TabContent
        id="id"
        selection_key="second"
      >
        <div
          aria-labelledby="id-tab-second"
          className="dnb-tabs__content"
          id="id-content-second"
          role="tabpanel"
        >
          <h2>
            Second
          </h2>
        </div>
      </TabContent>
    </div>
  </Wrapper>
</Tabs>
`;

exports[`Tabs scss have to match snapshot 1`] = `
"/*
* DNB Tabs
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * CSS Properties
 *
 */
:root {
  --font-family-std: 'Fedra Sans Std', sans-serif;
  --font-family-demi: 'Fedra Sans Std Demi', sans-serif;
  --color-sea-green-alt: #008484;
  --color-mint-green-50: #d2f0e9;
  --color-mint-green-25: #e9f8f4;
  --color-sea-green-4: #f5fafa;
  --color-sea-green-alt-30: #b3dada;
  --color-signal-yellow-30: #ffffd7;
  --color-cherry-red: #e10076;
  --color-cherry-red-80: #fdebf4;
  --color-black: #000;
  --color-black-80: #333;
  --color-black-30: #b3b3b3;
  --color-white: #fff;
  --color-black-border: #cdcdcd;
  --color-black-background: #fafafa;
  --color-sea-green: #007272;
  --color-mint-green: #a5e1d2;
  --color-summer-green: #28b482;
  --color-emerald-green: #14555a;
  --color-ocean-green: #00343e;
  --color-signal-orange: #ff5400;
  --color-signal-yellow: #ffff7a;
  --color-indigo: #23195a;
  --color-violet: #6e2382;
  --color-sky-blue: #4bbed2; }

/*
 * Utilities
 */
/*
 * Typography
 *
 * The font-family properties are defined in the properties.scss file, like:
 * --font-family-std:
 *
 */
.dnb-typo-std,
.dnb-typo-book {
  font-family: var(--font-family-std);
  font-weight: normal;
  font-style: normal; }

.dnb-typo-medium {
  font-family: var(--font-family-std);
  font-weight: 600;
  font-style: normal; }

.dnb-typo-demi {
  font-family: var(--font-family-demi);
  font-weight: normal;
  font-style: normal; }

.dnb-typo-number--lining {
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\"; }

.dnb-typo-number--tabular {
  font-variant-numeric: tabular-nums;
  font-feature-settings: \\"lnum\\"; }

.dnb-typo-number--oldstyle {
  font-variant-numeric: oldstyle-nums;
  font-feature-settings: \\"onum\\"; }

.dnb-typo-number--proportional {
  font-variant-numeric: proportional-nums;
  font-feature-settings: \\"pnum\\"; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Book.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Book.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal;
  font-display: fallback; }

@font-face {
  font-family: 'Fedra Sans Std';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Medium.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Medium.ttf\\") format(\\"truetype\\");
  font-weight: 600;
  font-style: normal;
  font-display: fallback; }

@font-face {
  font-family: 'Fedra Sans Std Demi';
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot\\");
  src: url(\\"../../../../assets/fonts//FedraSansStd-Demi.eot?#iefix\\") format(\\"embedded-opentype\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff2\\") format(\\"woff2\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.woff\\") format(\\"woff\\"), url(\\"../../../../assets/fonts//FedraSansStd-Demi.ttf\\") format(\\"truetype\\");
  font-weight: normal;
  font-style: normal;
  font-display: fallback; }

/*
 * Scopes
 * 
 */
.dnb-tabs {
  font-family: var(--font-family-std, sans-serif);
  font-size: 1rem;
  font-weight: normal;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\"; }
  .dnb-tabs *,
  .dnb-tabs ::before,
  .dnb-tabs ::after {
    background-repeat: no-repeat;
    box-sizing: border-box; }
  .dnb-tabs ::before,
  .dnb-tabs ::after {
    text-decoration: inherit;
    vertical-align: inherit; }

/*
 * Tabs component
 *
 */
.dnb-tabs__tabs {
  position: relative;
  z-index: 1;
  display: flex;
  flex-direction: row;
  align-content: flex-start;
  justify-content: space-between;
  align-items: center;
  flex-wrap: nowrap; }
  .dnb-tabs__tabs__tablist {
    display: flex;
    flex-direction: row; }
  .dnb-tabs__tabs--left .dnb-tabs__tabs__tablist {
    justify-content: flex-start; }
  .dnb-tabs__tabs--right .dnb-tabs__tabs__tablist {
    justify-content: flex-end; }
  .dnb-tabs__tabs--center .dnb-tabs__tabs__tablist {
    justify-content: center; }
  .dnb-tabs__tabs::after {
    content: '';
    position: absolute;
    z-index: -1;
    left: -50vw;
    right: -50vw;
    bottom: 0;
    width: 200vw;
    height: 1px; }
  .dnb-tabs__tabs .dnb-tablink {
    position: relative;
    z-index: 1;
    margin: 0 0 0 2.5em;
    padding: 0.5em 0;
    background-color: transparent;
    border-bottom: 1px solid transparent;
    cursor: pointer;
    outline: none;
    border: none; }
    .dnb-tabs__tabs .dnb-tablink-title {
      font-size: 1em;
      line-height: 1.5em;
      white-space: nowrap; }
    .dnb-tabs__tabs .dnb-tablink > span[hidden] {
      display: flex;
      flex-direction: column;
      visibility: hidden;
      height: 0;
      overflow: hidden;
      letter-spacing: 2px; }
    .dnb-tabs__tabs .dnb-tablink::after {
      content: '';
      position: absolute;
      z-index: -1;
      left: 0;
      right: 0;
      bottom: 0;
      width: 100%;
      height: 1px; }
  .dnb-tabs__tabs .dnb-tablink:first-child {
    margin-left: 0; }
  @media screen and (max-width: 700px) {
    .dnb-tabs__tabs__tablist {
      display: block; }
    .dnb-tabs__tabs .dnb-tablink {
      margin-top: 0.5em;
      margin-left: 0; } }

.dnb-tabs__content {
  padding: 1rem 0.5rem; }
"
`;
